st2common/st2common/models/api/base.py

```diff
@@ -245,16 +243,6 @@ def cast_func(value):
                     except Exception as e:
                         raise exc.HTTPInternalServerError(detail=e.message,
                                                           comment=traceback.format_exc())
-
-                    # Set default pack if one is not provided for resource create
-                    if function_name == 'post' and not hasattr(obj, 'pack'):
-                        extra = {
-                            'resource_api': obj,
-                            'default_pack_name': DEFAULT_PACK_NAME
-                        }
-                        LOG.debug('Pack not provided in the body, setting a default pack name',
-                                  extra=extra)
-                        setattr(obj, 'pack', DEFAULT_PACK_NAME)
```

Comment:
I moved this logic to https://github.com/StackStorm/st2/pull/2842/files#diff-530450a16546013ae50b71cd07dcb2b4R111 to avoid it leaking into every post request. I'm not entirely certain we don't rely on it anywhere else. I'm not sure we should have ever done anything like that in the first place. I'm absolutely positive that this is one of the worst places it could have been done.

@kami thoughts? https://github.com/StackStorm/st2/commit/39e5ac529b10cbc25bd1c4ddf58fac312bf33f20